nautobot/core/celery/__init__.py

```diff
@@ -53,60 +43,74 @@ def register_task(self, task, **options):
 
 
 @signals.import_modules.connect
-def import_jobs_as_celery_tasks(sender, database_ready=True, **kwargs):
+def import_jobs(sender=None, **kwargs):
     """
-    Import system Jobs into Celery as well as Jobs from JOBS_ROOT and GIT_ROOT.
+    Import system Jobs into Nautobot as well as Jobs from JOBS_ROOT and GIT_ROOT.
+
+    Note that app-provided jobs are automatically imported at startup time via NautobotAppConfig.ready()
+    """
+    import nautobot.core.jobs  # noqa: F401
+
+    _import_jobs_from_jobs_root()
 
-    Note that app-provided Jobs are automatically imported at startup time via NautobotAppConfig.ready()
+    try:
+        _import_jobs_from_git_repositories()
+    except ProgrammingError:  # Database not ready yet, as may be the case on initial startup and migration
+        pass
+
+
+def _import_jobs_from_jobs_root():
+    """
+    (Re)import all modules in settings.JOBS_ROOT.
     """
-    logger.debug("Importing system Jobs")
-    sender.loader.import_task_module("nautobot.core.jobs")
-
-    jobs_root = settings.JOBS_ROOT
-    if jobs_root and os.path.exists(jobs_root):
-        if jobs_root not in sys.path:
-            sys.path.append(jobs_root)
-        for _, module_name, _ in pkgutil.iter_modules([jobs_root]):
-            try:
-                logger.debug("Importing Jobs from %s in JOBS_ROOT", module_name)
-                existing_module = find_spec(module_name)
-                if existing_module is not None:
-                    existing_module_path = os.path.realpath(existing_module.origin)
-                    jobs_root_path = os.path.realpath(jobs_root)
-                    if not existing_module_path.startswith(jobs_root_path):
-                        raise ImportError(
-                            f"JOBS_ROOT Jobs module {module_name} conflicts with existing module {existing_module_path}"
-                        )
-                sender.loader.import_task_module(module_name)
-            except Exception as exc:
-                logger.exception(exc)
-
-    git_root = settings.GIT_ROOT
-    if git_root and os.path.exists(git_root):
-        if git_root not in sys.path:
-            sys.path.append(git_root)
-
-        # We can't detect which Git directories we're *supposed* to auto-load Jobs from if we can't read GitRepository
-        # records from the DB, unfortunately.
-        # We work around this in JobModel.job_task to try later loading Git jobs on-the-fly if needed.
-        if database_ready:
+    if not (settings.JOBS_ROOT and os.path.isdir(settings.JOBS_ROOT)):
```

Comment:
nitpick
```suggestion
    if not all((settings.JOBS_ROOT, os.path.isdir(settings.JOBS_ROOT))):
```

Reply:
Just to explain myself, I was traumatized by perl's `unless` statement. What does `unless apple is fruit or apple is vegetable` mean? 🤷

Reply:
I was hoping the parens were clear enough :) But I don't mind making this change either.

Reply:
Like I said, it's a nitpick so I'm fine either way

Reply:
Actually the change won't work since `all()` doesn't short-circuit logic:

```python
>>> JOBS_ROOT = None
>>> import os.path
>>> bool(JOBS_ROOT and os.path.isdir(JOBS_ROOT))
False
>>> bool(all((JOBS_ROOT, os.path.isdir(JOBS_ROOT))))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/Cellar/python@3.9/3.9.16/Frameworks/Python.framework/Versions/3.9/lib/python3.9/genericpath.py", line 42, in isdir
    st = os.stat(s)
TypeError: stat: path should be string, bytes, os.PathLike or integer, not NoneType
```